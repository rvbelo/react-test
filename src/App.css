#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}
form {
  background-color: #666;
}

form span {
  flex: 1;
  color: #90efdb;
  height: 1.8rem;
  font-size: 1.5rem;
}

.field-cep {
  border: 1rem;
  background-color: burlywood;
  color: black;
}

.field-cep label {
  font-size: 1.2rem;
  font-variant: small-caps;
  margin: .8rem;
}

.field-cep input {
  height: 2rem;
  background: #333;
  color: white;
  font-size: 1.8rem;
}
