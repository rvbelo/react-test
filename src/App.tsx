import { useEffect, useState } from 'react'
import { Axios } from 'axios'
import './App.css'

const axiosClient = new Axios({
  baseURL: 'https://viacep.com.br/ws/',
  headers: {
    "Accept": 'application/json',
    "Content-Type": 'application/json'
  }
})

function App() {
  const [ address, setAddress ] = useState('')

  const handleSearch = (cep: string) => {
    if(cep){
      axiosClient.get(`${cep}/json/`)
        .then(response => {
          setAddress(response.data)
        })
    }
  }

  useEffect(()=>{
    console.log(address)
  }, [address])

  return (
    <>
      <form>
        <fieldset className='field-cep'>
          <label htmlFor='cep'>Cep</label>
          <input className='input-cep' type='text' id='cep' name='cep' onBlur={(e) => {handleSearch(e.target.value)}}/>
        </fieldset>
        <span>Resultado: {address}</span>
      </form>
    </>
  )
}

export default App
