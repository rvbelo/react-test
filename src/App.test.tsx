import React from 'react'
import renderer from 'react-test-renderer'
import {render, screen} from '@testing-library/react';
import { expect, test } from 'vitest'
import App from './App'

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

test('Form page must be rendered correctly', () => {
  const component = renderer.create(
    <App />,
  )
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()

  // re-rendering
  tree = toJson(component)
  expect(tree).toMatchSnapshot()

  // re-rendering
  tree = toJson(component)
  expect(tree).toMatchSnapshot()

  
})

test('Input CEP must be in document', () => {
    const component = renderer.create(<App />)

    const testInstance = component.root
    expect(testInstance.findByProps({className: 'input-cep'})).toBeDefined()
})
