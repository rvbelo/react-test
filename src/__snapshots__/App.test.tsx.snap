// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form page must be rendered correctly 1`] = `
<form>
  <fieldset
    className="field-cep"
  >
    <label
      htmlFor="cep"
    >
      Cep
    </label>
    <input
      className="input-cep"
      id="cep"
      name="cep"
      onBlur={[Function]}
      type="text"
    />
  </fieldset>
  <span>
    Resultado: 
  </span>
</form>
`;

exports[`Form page must be rendered correctly 2`] = `
<form>
  <fieldset
    className="field-cep"
  >
    <label
      htmlFor="cep"
    >
      Cep
    </label>
    <input
      className="input-cep"
      id="cep"
      name="cep"
      onBlur={[Function]}
      type="text"
    />
  </fieldset>
  <span>
    Resultado: 
  </span>
</form>
`;

exports[`Form page must be rendered correctly 3`] = `
<form>
  <fieldset
    className="field-cep"
  >
    <label
      htmlFor="cep"
    >
      Cep
    </label>
    <input
      className="input-cep"
      id="cep"
      name="cep"
      onBlur={[Function]}
      type="text"
    />
  </fieldset>
  <span>
    Resultado: 
  </span>
</form>
`;
